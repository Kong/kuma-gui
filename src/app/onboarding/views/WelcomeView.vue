<template>
  <RouteView>
    <RouteTitle
      :title="t('onboarding.routes.welcome.title', {name: t('common.product.name')})"
    />
    <AppView>
      <div>
        <OnboardingPage>
          <template #header>
            <OnboardingHeading>
              <template #title>
                Welcome to {{ PRODUCT_NAME }}
              </template>

              <template #description>
                <p>
                  Congratulations on downloading {{ PRODUCT_NAME }}! You are just a <strong>few minutes</strong> away from getting your service mesh fully online.
                </p>

                <p>
                  We have automatically detected that you are running on <strong>{{ enviromentFormatted }}</strong>.
                </p>
              </template>
            </OnboardingHeading>
          </template>

          <template #content>
            <h2 class="text-center">
              Letâ€™s get started:
            </h2>

            <div class="item-status-list-wrapper">
              <ul class="item-status-list">
                <li
                  v-for="item in statuses"
                  :key="item.name"
                >
                  <span class="circle mr-2">
                    <KIcon
                      v-if="item.status"
                      icon="check"
                      size="14"
                      color="currentColor"
                    />
                  </span>

                  {{ item.name }}
                </li>
              </ul>
            </div>
          </template>

          <template #navigation>
            <OnboardingNavigation next-step="onboarding-deployment-types" />
          </template>
        </OnboardingPage>

        <WelcomeAnimationSvg :longer="isMulticluster" />
      </div>
    </AppView>
  </RouteView>
</template>

<script lang="ts" setup>
import { KIcon } from '@kong/kongponents'
import { computed } from 'vue'

import OnboardingHeading from '../components/OnboardingHeading.vue'
import OnboardingNavigation from '../components/OnboardingNavigation.vue'
import OnboardingPage from '../components/OnboardingPage.vue'
import WelcomeAnimationSvg from '../components/WelcomeAnimationSvg.vue'
import AppView from '@/app/application/components/app-view/AppView.vue'
import RouteTitle from '@/app/application/components/route-view/RouteTitle.vue'
import RouteView from '@/app/application/components/route-view/RouteView.vue'
import { PRODUCT_NAME } from '@/constants'
import { useStore } from '@/store/store'
import { useI18n } from '@/utilities'

const store = useStore()
const { t } = useI18n()

const enviromentFormatted = computed(() => {
  const environment = store.getters['config/getEnvironment']
  return environment.charAt(0).toUpperCase() + environment.slice(1)
})

const isMulticluster = computed(() => store.getters['config/getMulticlusterStatus'])
const statuses = computed(() => [
  {
    name: `Run ${PRODUCT_NAME} control plane`,
    status: true,
  },
  {
    name: 'Learn about deployments',
    status: false,
  },
  {
    name: 'Learn about configuration storage',
    status: false,
  },
  ...isMulticluster.value ? [{ name: 'Add zones', status: false }] : [],
  {
    name: 'Create the mesh',
    status: false,
  },
  {
    name: 'Add services',
    status: false,
  },
  {
    name: 'Go to the dashboard',
    status: false,
  },
])
</script>

<style lang="scss" scoped>
.item-status-list-wrapper {
  display: flex;
  justify-content: center;
}

.item-status-list {
  margin-top: var(--spacing-md);
}

.item-status-list > * + * {
  margin-top: var(--spacing-xs);
}

.circle {
  height: 1rem;
  width: 1rem;
  border-radius: 50%;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  background-color: var(--grey-300);
}
</style>
