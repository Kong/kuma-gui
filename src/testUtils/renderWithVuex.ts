import { render, RenderOptions, RenderResult } from '@testing-library/vue'
import merge from 'lodash/merge'
import { StoreOptions } from 'vuex'
import Vue, { ComponentOptions } from 'vue'
import Store from '@/store'

function renderWithVuex(
  Component: typeof Vue | ComponentOptions<Vue>,
  options: RenderOptions<Vue> = { store: {}, routes: [] },
): RenderResult {
  const store: StoreOptions<any> = Store()

  // Render the component and merge the original store and the custom one
  // provided as a parameter. This way, we can alter some behaviors of the
  // initial implementation.

  return render(Component, { ...options, store: merge(store, options.store), routes: options.routes })
}

export default renderWithVuex
