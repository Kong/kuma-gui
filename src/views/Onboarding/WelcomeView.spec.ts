import WelcomeView from './WelcomeView.vue'
import renderWithVuex from '@/testUtils/renderWithVuex'

describe('WelcomeView.vue', () => {
  const customStore = {
    modules: { config: { state: { tagline: 'Kuma', clientConfig: { environment: 'universal' } } } },
  }

  it('renders snapshot', () => {
    const { container } = renderWithVuex(WelcomeView, { store: customStore, routes: [] })

    expect(container).toMatchSnapshot()
  })

  it('renders Kubernetess', () => {
    const { getByText } = renderWithVuex(WelcomeView, {
      routes: [],
      store: { modules: { config: { state: { clientConfig: { mode: 'global', environment: 'kubernetess' } } } } },
    })

    expect(getByText(/Kubernetess/)).toBeInTheDocument()
  })
})
