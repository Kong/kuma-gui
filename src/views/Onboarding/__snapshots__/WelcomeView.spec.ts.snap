// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeView.vue renders snapshot 1`] = `
<div>
  <div
    class="screen"
  >
    <div
      class="welcome-container"
    >
      <div
        class="content"
      >
        <h1
          class="welcome-title"
        >
           Welcome to Kuma
        </h1>
        <p
          class="welcome-description"
        >
           Congratulations on downloading Kuma! You are just a 
          <strong>
            few minutes
          </strong>
           away from getting your service mesh fully online. 
        </p>
        <p
          class="welcome-description"
        >
           We have automatically detected that you are running on 
          <strong>
            Universal
          </strong>
          . 
        </p>
        <h2
          class="welcome-detected"
        >
           Let's get started: 
        </h2>
        <ul>
          
          <li
            class="flex items-center mb-2"
          >
            <span
              class="circle"
            >
              <span
                class="kong-icon-check kong-icon"
                data-v-637de6fa=""
              >
                <svg
                  fill="none"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  
  
                  <path
                    d="m14 7-6 6-3-3"
                    fill="none"
                    stroke="#A3BBCC"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2.25"
                  />
                  

                </svg>
                

              </span>
            </span>
             Run Kuma control plane
          </li>
          <li
            class="flex items-center mb-2"
          >
            <span
              class="circle"
            >
              <!--v-if-->
            </span>
             Learn about deployments
          </li>
          <li
            class="flex items-center mb-2"
          >
            <span
              class="circle"
            >
              <!--v-if-->
            </span>
             Learn about configuration storage
          </li>
          <li
            class="flex items-center mb-2"
          >
            <span
              class="circle"
            >
              <!--v-if-->
            </span>
             Create the mesh
          </li>
          <li
            class="flex items-center mb-2"
          >
            <span
              class="circle"
            >
              <!--v-if-->
            </span>
             Add services
          </li>
          <li
            class="flex items-center mb-2"
          >
            <span
              class="circle"
            >
              <!--v-if-->
            </span>
             Go to the dashboard
          </li>
          
        </ul>
      </div>
      <div
        class="welcome-navigation"
      >
        <div
          class="mt-4 flex items-center flex-col sm:flex-row justify-end"
        >
          <!--v-if-->
          <div>
            <button
              class="small rounded btn-link k-button skip-button"
              data-v-29036f3a=""
              type="button"
            >
              
              <!---->
              
              
               Skip Setup 
              
              <!---->
            </button>
            <span
              class="inline-block"
            >
              <a
                class="medium rounded primary k-button navigation-button navigation-button--next"
                data-v-29036f3a=""
                type="button"
              >
                
                <!---->
                
                
                Next
                
                <!---->
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
    <svg
      class="svg"
      longer="false"
      viewBox="0 0 1920 1080"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <defs>
        <lineargradient
          id="a"
          x1="64.8%"
          x2="34.4%"
          y1="-12.6%"
          y2="153.1%"
        >
          <stop
            offset="0%"
            stop-color="#260D50"
          />
          <stop
            offset="100%"
            stop-color="#822DC5"
          />
        </lineargradient>
        <lineargradient
          id="b"
          x1="0%"
          x2="107.4%"
          y1="41.5%"
          y2="41.5%"
        >
          <stop
            offset="0%"
            stop-color="#260D50"
          />
          <stop
            offset="100%"
            stop-color="#822DC5"
          />
        </lineargradient>
      </defs>
      <g
        fill="none"
        fill-rule="evenodd"
        mask="url(#ccard)"
        transform="translate(-63 -285)"
      >
        <g
          stroke="#822DC5"
          stroke-width="6"
        >
          <path
            class="nodepath"
            d="M1444 893h252"
          />
          <path
            class="nodepath"
            d="M1529 705h232M1452 603h237"
            stroke-opacity=".4"
          />
          <path
            class="nodepath"
            d="M1754 563l-332 332h-76M1444 935l121 121M263 859l156 156"
          />
          <path
            class="nodepath"
            d="M742 781H556"
            stroke-opacity=".4"
          />
          <path
            class="nodepath"
            d="M697 736H513"
          />
          <path
            class="nodepath"
            d="M695 783V577"
            stroke-opacity=".4"
          />
          <path
            class="nodepath"
            d="M261 1026V751M509 573V438M1502 415l291 290"
          />
          <path
            class="nodepath"
            d="M698 912L26 240M1368 411v540l61 61 95-95M1434 3h267l193 192v354"
            stroke-opacity=".4"
          />
          <path
            class="nodepath"
            d="M517 411h342l138 138M1416 573v242l371 323"
          />
          <path
            class="nodepath"
            d="M1486 817V612l-146-146M839 243h-97l-83 84v348M1698 1063V817l58-57h122M1069 299L558 810M696 1058H585L468 941V570L322 424"
          />
          <path
            class="nodepath"
            d="M277 528l160 160 236-236 121 121M632 979h-45l-67-67v-86H0M106 669h275M70 707h331M207 745h210M85 784h356M1417 558h228M1609 634h203M528 946h76M619 604v131M1359 567l125 125M1332 594l156 156M1594 1070V959M381 632L260 753"
          />
          <path
            class="nodepath"
            d="M605 817V497M1851 959h-518M944 570H390"
            stroke-opacity=".4"
          />
          <path
            class="nodepath"
            d="M638 912H342M635 1139V912"
          />
          <path
            class="final"
            d="M1024 573h297v532h-31"
            stroke="url(#a)"
            transform="translate(0 3)"
          />
          <path
            class="final circle"
            d="M1024 592a18 18 0 100-36 18 18 0 000 36z"
            fill="url(#b)"
            fill-rule="nonzero"
            transform="translate(0 3)"
          />
        </g>
        <foreignobject
          height="100%"
          width="100%"
          x="0"
          y="0"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </foreignobject>
      </g>
    </svg>
  </div>
</div>
`;
